### PR TITLE
add ownerCount method

### PR DESCRIPTION
**Pull request by wilsoncusack**
Convenience method https://github.com/cantinasec/review-coinbase-smartwallet/issues/5

### CODE REVIEW

File: src/MultiOwnable.sol [hunk 1]
```diff
@@ -137,9 +136,8 @@ contract MultiOwnable {
     /// @param index The index of the owner to be removed.
     /// @param owner The ABI encoded bytes of the owner to be removed.
     function removeLastOwner(uint256 index, bytes calldata owner) external virtual onlyOwner {
-        MultiOwnableStorage storage $ = _getMultiOwnableStorage();
-        uint256 ownersRemaining = $.nextOwnerIndex - $.removedOwnersCount;
-        if (ownersRemaining > 1) {
+        uint256 ownersRemaining = ownerCount();
+        if (ownerCount() > 1) {
```
Comment (optimumsec) [inline src/MultiOwnable.sol:140]: No need to call `ownerCount()` twice here @wilsoncusack

File: src/MultiOwnable.sol [hunk 2]
```diff
@@ -137,8 +136,7 @@ contract MultiOwnable {
     /// @param index The index of the owner to be removed.
     /// @param owner The ABI encoded bytes of the owner to be removed.
     function removeLastOwner(uint256 index, bytes calldata owner) external virtual onlyOwner {
-        MultiOwnableStorage storage $ = _getMultiOwnableStorage();
-        uint256 ownersRemaining = $.nextOwnerIndex - $.removedOwnersCount;
+        uint256 ownersRemaining = ownerCount();
```
Comment (xenoliss) [inline src/MultiOwnable.sol:139]: I think we can get rid of this stack variable and do the same thing as in `removeOwnerAtIndex():

```solidity
if (ownerCount() > 1) {
    revert NotLastOwner(ownersRemaining);
}
```

Comment (wilsoncusack) [reply src/MultiOwnable.sol:139]: but then I would need to call `ownerCount` again here `revert NotLastOwner(ownersRemaining);`

Comment (wilsoncusack) [reply src/MultiOwnable.sol:139]: Would be `revert NotLastOwner(ownerCount());`

Comment (xenoliss) [reply src/MultiOwnable.sol:139]: Oh yes right, I missed that.

**Review by xenoliss — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for xenoliss @ 2024-04-19 17:06:50 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

File: src/MultiOwnable.sol (hunk 2 above)
Comment (wilsoncusack) [reply src/MultiOwnable.sol:139]: But yeah could debate: do we penalize gas on the non revert path in order to save on revert?

File: src/MultiOwnable.sol [hunk 3]
```diff
@@ -120,8 +120,7 @@ contract MultiOwnable {
     /// @param index The index of the owner to be removed.
     /// @param owner The ABI encoded bytes of the owner to be removed.
     function removeOwnerAtIndex(uint256 index, bytes calldata owner) external virtual onlyOwner {
```
Comment (pegahcarter) [inline src/MultiOwnable.sol:122]: These methods seem like overkill as they require the caller to pass in both the owner and index, when either should suffice.

**Pull request merged**